BepInEx.Core/Contract/Attributes.cs

```diff
@@ -57,7 +61,29 @@ internal static BepInPlugin FromCecilType(TypeDefinition td)
 			if (attr == null)
 				return null;
 
-			return new BepInPlugin((string)attr.ConstructorArguments[0].Value, (string)attr.ConstructorArguments[1].Value, (string)attr.ConstructorArguments[2].Value);
+			var guid = (string)attr.ConstructorArguments[0].Value;
+
+			if (attr.ConstructorArguments.Count == 1)
+			{
+				var assembly = td.Module.Assembly;
+				var name = assembly.Name;
+				SemVersion version;
+
+				var versionAttribute = assembly.GetCustomAttribute(typeof(AssemblyInformationalVersionAttribute).FullName);
+
+				if (versionAttribute != null && versionAttribute.ConstructorArguments.Count == 1)
+				{
+					SemVersion.TryParse(versionAttribute.ConstructorArguments.Single().Value as string, out version);
+				}
+				else
+				{
+					version = new SemVersion(name.Version);
+				}
+
+				return new BepInPlugin(guid, name.Name, version);
+			}
```

Comment:
Was this meant to be used as follows:

```cs
[BepInPlugin("my.plugin.id")]
class MyPlugin : BasePlugin {}
```
and it would grab plugin version and name from assembly name and version?

To me sounds like an interesting idea, but I'm not sure if it's good to grab the name from assembly name. Name was initially intended as a human-readable normal name of the plugin and not really an assembly name.

cc @bbepis for opinions on this

Reply:
I could just use [AssemblyTitleAttribute](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assemblytitleattribute?view=netframework-3.5) with fallback to assembly name, just like AssemblyInformationalVersionAttribute with assembly version